models/database.go

```diff
@@ -650,19 +650,65 @@ func SetupDB(sqlDB *sql.DB, params *SetupDBParams) (*reform.DB, error) {
 		latestVersion = *params.MigrationVersion
 	}
 	var currentVersion int
-	err := db.QueryRow("SELECT id FROM schema_migrations ORDER BY id DESC LIMIT 1").Scan(&currentVersion)
-	if pErr, ok := err.(*pq.Error); ok && pErr.Code == "42P01" { // undefined_table (see https://www.postgresql.org/docs/current/errcodes-appendix.html)
-		err = nil
+	errDB := db.QueryRow("SELECT id FROM schema_migrations ORDER BY id DESC LIMIT 1").Scan(&currentVersion)
+	if pErr, ok := errDB.(*pq.Error); ok && pErr.Code == "28000" { // invalid_authorization_specification	(see https://www.postgresql.org/docs/current/errcodes-appendix.html)
+		var databaseName = params.Username
+		var roleName = params.Username
+
+		if params.Logf != nil {
+			params.Logf("Creating database %s and role %s", databaseName, roleName)
+		}
+		db, err := OpenDB("127.0.0.1:5432", "", "postgres", "")
```

Comment:
Can we use the same credentials that we pass in flags?
Current logic might not work with HA planned in neat future.